### PR TITLE
Tarunyadav148

### PR DESCRIPTION
**Pull request by tarunyadav148**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

Added code for checking Magic Number

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2770

### CODE REVIEW

**Comment by tarunyadav148** [comment]
@bhav09 Can you please review this, sorry i forget to edit PR's title

File: C-Plus-Plus/math/magic_number.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    1234
+    Output:
+    1234 is Magic Number.
+2.
+    Input:
+    Enter the Number to check whether it is Magic Number or Not:
+    785
+    Output:
+    785 is Not Magic Number.
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/math/magic_number.cpp]: Add a new line at the EOF

File: C-Plus-Plus/math/magic_number.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/*
+A number is said to be a magic number,if the sum of its digits
+are calculated recursively till a single digit comes by adding 
+the sum of digits after every addition. 
+If single digit comes out to be 1 then the number is a Magic Number.
```
Comment (HarshCasper) [inline C-Plus-Plus/math/magic_number.cpp]: Use proper sentences

File: C-Plus-Plus/math/magic_number.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    Enter the Number to check whether it is Magic Number or Not:
+    1234
+    Output:
+    1234 is Magic Number.
+2.
+    Input:
+    Enter the Number to check whether it is Magic Number or Not:
+    785
+    Output:
+    785 is Not Magic Number.
```
Comment (HarshCasper) [inline C-Plus-Plus/math/magic_number.cpp]: Remove extra I/O samples

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by tarunyadav148** [comment]
I have done the changes please review the PR @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**